File: 2478-longest-nice-subarray/longest-nice-subarray.cpp
class Solution {
public:
    int longestNiceSubarray(vector<int>& nums) {
        int left=0;
        int maxsize=0;
        int sum=0;
        int n=nums.size();
        for(int right=0;right<n;right++){
            
           while((sum&nums[right])!=0)
           { sum^=nums[left];
            left++;
        }
        sum|=nums[right];
        maxsize=max(maxsize,right-left+1);
       }
        return maxsize;
    }
};

File: 2478-longest-nice-subarray/longest-nice-subarray.py
class Solution:
    def longestNiceSubarray(self, nums: List[int]) -> int:
        left=0
        n=len(nums)
        sum=0
        maxlen=0
        for right in range(n):
            while((sum&nums[right])!=0):
                sum^=nums[left]
                left+=1
            sum|=nums[right]
            maxlen=max(maxlen,right-left+1)
        return maxlen        
        